### PR TITLE
Login Page with Authentication

### LINKED ISSUE #4: Login Page
# Development of the Application's Login Page.

:point_right: **Layouts and Palette of colors and other stuffs on Figma**

image

Feel free to break it into React components. **What? Why not define all of the UI components first?** it will give us more experience of how built up our UI components.

### PR DESCRIPTION
**Pull request by hdev14**
It was created:
 - Login page
 - Authentication
 - Auth reducer and Actions
 - and Alert to info invalid credentials

image

### CODE REVIEW

**Comment by hdev14** [comment]
When approved it will be necessary to remove the conflicts manually. xD

File: src/layouts/Auth/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Header from '../../components/Header';
+
+const Auth = ({ children }) => (
+  <div>
+    <Header />
+    asdfs
```
Comment (FrankKleiton) [inline src/layouts/Auth/index.js]: Remover 'asdfs'

File: src/redux/auth/sagas.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import AuthTypes from './types';
+import { signInSuccess } from '.';
+
+function* signIn({ payload }) {
+  try {
+    const response = yield call(api.post, '/login', payload);
+    const { token } = response.data;
+    // Set token to requests
+    api.defaults.headers.Authorization = `Baerer ${token}`;
```
Comment (FrankKleiton) [inline src/redux/auth/sagas.js]: Bearer ${token}

File: src/redux/auth/sagas.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    history.push('/materials');
+  } catch (err) {
+    toast.error('E-mail ou senha incorretos');
+  }
+}
+
+function setToken({ payload }) {
+  if (payload.auth && payload.auth.token) {
+    // Set token to requests
+    api.defaults.headers.Authorization = `Baerer ${payload.auth.token}`;
```
Comment (FrankKleiton) [inline src/redux/auth/sagas.js]: Bearer ${payload.auth.token}

**Review by FrankKleiton — CHANGES_REQUESTED** [review]

File: src/layouts/Auth/index.js (hunk 1 above)
Comment (hdev14) [reply src/layouts/Auth/index.js]: kkk

**Review by FrankKleiton — APPROVED** [review]

**Pull request merged**